content/en/blog/_posts/2023-04-02-how-to-speed-up-pod-startup-from-kubelet-side.md

```diff
@@ -0,0 +1,56 @@
+
+We have mentioned some important factors here to consider from the kubelet's perspective, but this is not an exhaustive list. As Kubernetes v1.27 is set to release soon, this blog will highlight significant changes in v1.27 that aid in speeding up pod start-up.
+
+## Parallel Image Pulls
+
+Pulling images always takes some time and what's worse is that image pulls are done serially by default. In other words, kubelet will send only one image pull request to the image service at a time. Other image pull requests have to wait until the one being processed is complete.
+
+To enable parallel image pulls, set the `serializeImagePulls` field to false in the kubelet configuration. When `serializeImagePulls` is disabled, requests for image pulls are immediately sent to the image service and multiple images can be pulled concurrently.
+
+### Maximum parallel image pulls will help secure your node from overloading on image pulling
+
+We are introducing a new feature in kubelet that sets a limit on the number of parallel image pulls at the node level. This limit restricts the maximum number of images that can be pulled simultaneously. If there is an image pull request beyond this limit, it will be blocked until one of the ongoing image pulls finishes. Before enabling this feature, please ensure that your container runtime's image service can handle parallel image pulls effectively.
+
+To limit the number of simultaneous image pulls, you can configure the `maxParallelImagePulls` field in kubelet. By setting `maxParallelImagePulls` to a value of _n_, only _n_ images will be pulled concurrently. Any additional image pulls beyond this limit will wait until at least one ongoing pull is complete.
+
+More detials can be found in the KEP <https://kep.k8s.io/3673>.
+
+## bump default API QPS limits for Kubelet
+
+To improve pod startup in scenarios with multiple pods on a node, particularly sudden scaling situations, it is necessary for Kubelet to synchronize the pod status and prepare configmaps, secrets or volumes. This requires a large bandwidth to access kube-apiserver.
+
+In versions prior to v1.27, the default `kubeAPIQPS` was 5 and `kubeAPIBurst` was 10. However, Kubelet has increased these defaults to 50 and 100 respectively for better performance during pod startup. It's worth noting that this isn't the only reason why we've bumped up the API QPS limits for Kubelet.
+
+1. It has a potential to be hugely throttled now (default QPS = 5)
+2. In large clusters they can generate significant load anyway as there are a lot of them
+3. They have a dedicated PriorityLevel and FlowSchema that we can easily control
+
+In old days, we always encounter `volume mount timeout` on kubelet in node with more than 50 pods during pod start up. We suggest our customers to bump 3 or 4 times of kube api QPS and especially with customers that is using bare-metal(physical machines).
```

Comment:
"3 or 4 times" compared to the old default 5, or the new default 50? Should we make it more clear here?